src/key.rs

```diff
@@ -1,41 +1,54 @@
 use core::fmt;
 
 use crate::{
-    accumulator::accumulated_map::InputAccumulatedValues,
     cycle::CycleRecoveryStrategy,
     ingredient::MaybeChangedAfter,
     zalsa::{IngredientIndex, Zalsa},
     Database, Id,
 };
 
+// ANCHOR: DatabaseKeyIndex
 /// An integer that uniquely identifies a particular query instance within the
-/// database. Used to track output dependencies between queries. Fully ordered and
-/// equatable but those orderings are arbitrary, and meant to be used only for
-/// inserting into maps and the like.
+/// database. Used to track input and output dependencies between queries. Fully
+/// ordered and equatable but those orderings are arbitrary, and meant to be used
+/// only for inserting into maps and the like.
 #[derive(Copy, Clone, PartialEq, Eq, Hash)]
-pub struct OutputDependencyIndex {
+pub struct DatabaseKeyIndex {
```

Comment:
What is the motivation behind discarding the type level differentiation of input and output keys btw?

Reply:
I'd have to take a closer look but I think the reason is simply that they're the same now? see https://github.com/salsa-rs/salsa/issues/597